sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JacksonParser.scala

```diff
@@ -129,13 +129,14 @@ object JacksonParser extends Logging {
       case (VALUE_STRING, FloatType) =>
         // Special case handling for NaN and Infinity.
         val value = parser.getText
-        val lowerCaseValue = value.toLowerCase()
-        if (lowerCaseValue.equals("nan") ||
-          lowerCaseValue.equals("infinity") ||
-          lowerCaseValue.equals("-infinity") ||
-          lowerCaseValue.equals("inf") ||
-          lowerCaseValue.equals("-inf")) {
+        if (value.equals("NaN") ||
+          value.equals("Infinity") ||
+          value.equals("-Infinity")) {
           value.toFloat
+        } else if (value.equals("+INF")) {
```

Comment:
think from an end user point of view, we should be more consistent with ourselves, not some random rule by jackson.